tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs

```diff
@@ -148,6 +144,7 @@ public async Task NoImdsCancellation_UserCancelled_Async()
                 .ConfigureAwait(false);
 
             Assert.AreEqual(MsalError.RegionDiscoveryFailed, ex.ErrorCode);
+            regionDiscoveryProvider.clearStaticCache();
```

Comment:
Should there be some additional validation to confirm that the cache is clear?
Is there a separate test for that?

Reply:
I added that method specifically to clear the s_region static variable that holds the value for region. The value for that variable was persisting between tests and causing the tests to fail. No validation is required here.

Reply:
is there a test which tests the following:
- create one cca.AquireTokenForClient(scopes).WithAzureRegion(true);
and then same cca....
- cca.AcquireTokenForClient(scopes).WithAzureRegion(false)?
because that should show the 2nd call going to the global endpoint, correct?

---
In reply to: [544648991](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2308#discussion_r544648991) [](ancestors = 544648991)